tensorflow_addons/losses/contrastive_test.py

```diff
@@ -121,19 +96,49 @@ def test_non_default_margin(self):
         loss = self.evaluate(loss)
         self.assertAllClose(loss, 1.623333)
 
-    def test_no_reduction(self):
-        cl_obj = contrastive.ContrastiveLoss(reduction=tf.keras.losses.Reduction.NONE)
-        y_true = tf.constant([0, 0, 1, 1, 0, 1], dtype=tf.dtypes.int64)
-        y_pred = tf.constant([0.1, 0.3, 1.3, 0.7, 1.1, 0.5], dtype=tf.dtypes.float32)
-        loss = cl_obj(y_true, y_pred)
 
-        # Loss = y * (y`)^2 + (1 - y) * (max(m - y`, 0))^2
-        #      = [max(1 - 0.1, 0)^2, max(1 - 0.3, 0)^2,
-        #         1.3^2, 0.7^2, max(1 - 1.1, 0)^2, 0.5^2]
-        #      = [0.9^2, 0.7^2, 1.3^2, 0.7^2, 0^2, 0.5^2]
-        #      = [0.81, 0.49, 1.69, 0.49, 0, 0.25]
+def test_scalar_weighted():
+    cl_obj = contrastive.ContrastiveLoss()
+    y_true = tf.constant([0, 0, 1, 1, 0, 1], dtype=tf.dtypes.int64)
+    y_pred = tf.constant([0.1, 0.3, 1.3, 0.7, 1.1, 0.5], dtype=tf.dtypes.float32)
+    loss = cl_obj(y_true, y_pred, sample_weight=6.0)
 
-        self.assertAllClose(loss, [0.81, 0.49, 1.69, 0.49, 0.0, 0.25])
+    # Loss = y * (y`)^2 + (1 - y) * (max(m - y`, 0))^2
+    #      = [max(1 - 0.1, 0)^2, max(1 - 0.3, 0)^2,
+    #         1.3^2, 0.7^2, max(1 - 1.1, 0)^2, 0.5^2]
+    #      = [0.9^2, 0.7^2, 1.3^2, 0.7^2, 0^2, 0.5^2]
+    #      = [0.81, 0.49, 1.69, 0.49, 0, 0.25]
+    # Weighted loss = [0.81 * 6, 0.49 * 6, 1.69 * 6,
+    #                  0.49 * 6, 0 * 6, 0.25 * 6]
+    # Reduced loss = (0.81 + 0.49 + 1.69 + 0.49 + 0 + 0.25) * 6 / 6
+    #              = 3.73
+
+    np.testing.assert_allclose(loss, 3.73)
+
+
+def test_zero_weighted():
+    cl_obj = contrastive.ContrastiveLoss()
+    y_true = tf.constant([0, 0, 1, 1, 0, 1], dtype=tf.dtypes.int64)
+    y_pred = tf.constant([0.1, 0.3, 1.3, 0.7, 1.1, 0.5], dtype=tf.dtypes.float32)
+    loss = cl_obj(y_true, y_pred, sample_weight=0.0)
+    np.testing.assert_allclose(loss, 0.0)
+
+
+def test_no_reduction():
+    cl_obj = contrastive.ContrastiveLoss(reduction=tf.keras.losses.Reduction.NONE)
+    y_true = tf.constant([0, 0, 1, 1, 0, 1], dtype=tf.dtypes.int64)
+    y_pred = tf.constant([0.1, 0.3, 1.3, 0.7, 1.1, 0.5], dtype=tf.dtypes.float32)
+    loss = cl_obj(y_true, y_pred)
+
+    # Loss = y * (y`)^2 + (1 - y) * (max(m - y`, 0))^2
+    #      = [max(1 - 0.1, 0)^2, max(1 - 0.3, 0)^2,
+    #         1.3^2, 0.7^2, max(1 - 1.1, 0)^2, 0.5^2]
+    #      = [0.9^2, 0.7^2, 1.3^2, 0.7^2, 0^2, 0.5^2]
+    #      = [0.81, 0.49, 1.69, 0.49, 0, 0.25]
+
+    np.testing.assert_array_almost_equal(
```

Comment:
If it doesn't work with `np.testing.asser_allclose` it may be because self.assertAllClose and `np.testing.assert_allclose` don't have the same default `atol` and `rtol`. You just have to change those parameters to match the ones in `self.assertAllclose`.

Reply:
Thanks @gabrieldemarmiesse, since the test is about `Reduction.NONE`, which output is a list/array. What do you think?

Reply:
I'm not sure I understand what you mean. Can you detail why you chose `assert_array_almost_equal` instead of `assert_allclose`?

Reply:
I corrected it by `assert_allclose`. Thanks for the reminder, @gabrieldemarmiesse.